src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/IActionResultExecutor.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    public interface IActionResultExecutor<TResult> where TResult : IActionResult
+    {
+        Task ExecuteAsync(ActionContext context, TResult result);
```

Comment:
Most of the executors we have already follow this pattern, I'm just adding an interface.

I have big plans for these..... (trails off in a sinister way)